modules/rank-eval/src/main/java/org/elasticsearch/index/rankeval/PrecisionAtN.java

```diff
@@ -78,24 +81,31 @@ public int getN() {
         return n;
     }
 
+    private static final ParseField SIZE_FIELD = new ParseField("size");
+    private static final ConstructingObjectParser<PrecisionAtN, ParseFieldMatcherSupplier> PARSER = new ConstructingObjectParser<>(
+            "precision_at", a -> new PrecisionAtN((Integer) a[0]));
+
+    static {
+        PARSER.declareInt(ConstructingObjectParser.constructorArg(), SIZE_FIELD);
+    }
+
+    public static PrecisionAtN fromXContent(XContentParser parser, ParseFieldMatcherSupplier matcher) {
+        return PARSER.apply(parser, matcher);
+    }
+
     /** Compute precisionAtN based on provided relevant document IDs.
      * @return precision at n for above {@link SearchResult} list.
      **/
     @Override
-    public EvalQueryQuality evaluate(SearchHit[] hits, RatedQuery intent) {
-        Map<String, Integer> ratedDocIds = intent.getRatedDocuments();
-        
-        Collection<String> relevantDocIds = new ArrayList<>(); 
-        for (Entry<String, Integer> entry : ratedDocIds.entrySet()) {
-            if (Rating.RELEVANT.equals(RatingMapping.mapTo(entry.getValue()))) {
-                relevantDocIds.add(entry.getKey());
-            }
-        }
-        
-        Collection<String> irrelevantDocIds = new ArrayList<>(); 
-        for (Entry<String, Integer> entry : ratedDocIds.entrySet()) {
-            if (Rating.IRRELEVANT.equals(RatingMapping.mapTo(entry.getValue()))) {
-                irrelevantDocIds.add(entry.getKey());
+    public EvalQueryQuality evaluate(SearchHit[] hits, List<RatedDocument> ratedDocs) {
+
+        Collection<String> relevantDocIds = new ArrayList<>();
+        Collection<String> irrelevantDocIds = new ArrayList<>();
+        for (RatedDocument doc : ratedDocs) {
+            if (Rating.RELEVANT.equals(RatingMapping.mapTo(doc.getRating()))) {
```

Comment:
I'm not sure I'm totally happy with current mapping from the enum to the rating value. Relevant/Irrelevant or 0/1 works well for precion@ but probably not for something like Discounted Cumulative Gain, so maybe we need an abstraction for this.